### PR TITLE
Added mysql support

### PR DESCRIPTION
**Pull request by MelonSmasher**
Added support for JIRA/Confluence instances using MySQL DB backends.

### CODE REVIEW

File: bin/atlassian-backup [hunk 1]
```diff
@@ -61,7 +67,11 @@ function backup_attachments() {
 function dump_database() {
     echo "  - Dumping $NATURAL_TOOL_NAME database"
     DATABASE_DUMP_OUTPUT="${OUTPUT_PREFIX}-database-dump.sql.gz"
-    $PGENGINE/pg_dump "$DB_NAME" | gzip > "$DATABASE_DUMP_OUTPUT"
+    if [ "${DB_BACKEND}" = 'mysql' ]; then
+        $MYENGINE --single-transaction --host="$MYSQL_HOST" --user="$MYSQL_USER" --password="$MYSQL_PASS" "$DB_NAME" | gzip > "$DATABASE_DUMP_OUTPUT"
+    else
```
Comment (jmcgeheeiv) [inline bin/atlassian-backup]: elif [ "${DB_BACKEND}" = 'pg' ]; then

File: bin/atlassian-backup [hunk 2]
```diff
@@ -61,7 +67,11 @@ function backup_attachments() {
...
+        $PGENGINE/pg_dump "$DB_NAME" | gzip > "$DATABASE_DUMP_OUTPUT"
+    fi
```
Comment (jmcgeheeiv) [inline bin/atlassian-backup:83]: else
    echo "ERROR: No such DB_BACKEND '$DB_BACKEND'" 1>&2
    exit 2  # or whatever exit status you like
fi

File: configs/atlassian-backup.conf [hunk 3]
```diff
@@ -14,6 +17,18 @@ JIRA_BACKUP_DIR='/atlassian/backups/jira'
...
+JIRA_DB_BACKEND='pg'
+# JIRA_DB_BACKEND='mysql'
+# A database user that has access to your JIRA MySQL DB
+JIRA_MYSQL_USER=''
+# The password for the user that has access to your JIRA MySQL DB
+JIRA_MYSQL_PASS=''
+# The MySQL host that hosts your JIRA DB
+JIRA_MYSQL_HOST=''
 # JIRA Database
 JIRA_DB_NAME='jiradb_6_4'
```
Comment (jmcgeheeiv) [inline configs/atlassian-backup.conf]: I recommend (but I do not demand) removing "_6_4".  Database names in Atlassian documentation do not include the version, but I appended it anyway.

 As it turns out, it is unnatural to change the database name when upgrading, so Atlassian is right.

File: configs/atlassian-backup.conf [hunk 4]
```diff
@@ -27,5 +42,17 @@ CONFLUENCE_BACKUP_DIR='/atlassian/backups/confluence'
...
+CONFLUENCE_DB_BACKEND='pg'
+# CONFLUENCE_DB_BACKEND='mysql'
+# A database user that has access to your Confluence MySQL DB
+CONFLUENCE_MYSQL_USER=''
+# The password for the user that has access to your Confluence MySQL DB
+CONFLUENCE_MYSQL_PASS=''
+# The MySQL host that hosts your Confluence DB
+CONFLUENCE_MYSQL_HOST=''
 # Confluence Database
 CONFLUENCE_DB_NAME='confluencedb_5_7_1'
```
Comment (jmcgeheeiv) [inline configs/atlassian-backup.conf]: I recommend (but I do not demand) removing "_5_7_1". Database names in Atlassian documentation do not include the version, but I appended it anyway.

As it turns out, it is unnatural to change the database name when upgrading, so Atlassian is right.

**Review by jmcgeheeiv — CHANGES_REQUESTED** [review]
Thank you for this excellent enhancement.  I have migrated to Atlassian Cloud, so I am unable to test changes.  However it is evident that you can handle testing for me.

Please have a look at the results of my code review.

File: bin/atlassian-backup (hunk 1 above)
Comment (MelonSmasher) [reply bin/atlassian-backup]: I've added this in.

File: bin/atlassian-backup (hunk 2 above)
Comment (MelonSmasher) [reply bin/atlassian-backup:83]: I've added this in as well.

File: configs/atlassian-backup.conf (hunk 3 above)
Comment (MelonSmasher) [reply configs/atlassian-backup.conf]: I've removed the version numbers from the db names. I also updated the Jira and Confluence home paths to more recent defaults.

File: configs/atlassian-backup.conf (hunk 4 above)
Comment (MelonSmasher) [reply configs/atlassian-backup.conf]: I've removed the version numbers from the db names. I also updated the Jira and Confluence home paths to more recent defaults.

**Comment by MelonSmasher** [comment]
Hi @jmcgeheeiv,
I've implemented the changes you requested. I've tested this on Ubuntu 16.04.4 LTS running the latest versions of Jira and Confluence. It seems to work perfectly.

File: bin/atlassian-backup [hunk 5]
```diff
@@ -61,7 +73,14 @@ function backup_attachments() {
 function dump_database() {
     echo "  - Dumping $NATURAL_TOOL_NAME database"
     DATABASE_DUMP_OUTPUT="${OUTPUT_PREFIX}-database-dump.sql.gz"
-    $PGENGINE/pg_dump "$DB_NAME" | gzip > "$DATABASE_DUMP_OUTPUT"
+    if [ "${DB_BACKEND}" = 'mysql' ]; then
+        $MYENGINE --single-transaction --host="$MYSQL_HOST" --user="$MYSQL_USER" --password="$MYSQL_PASS" "$DB_NAME" | gzip > "$DATABASE_DUMP_OUTPUT"
```
Comment (jmcgeheeiv) [inline bin/atlassian-backup:77]: So sorry I forgot this last time:
$MYENGINE/mysqldump

**Review by jmcgeheeiv — APPROVED** [review]

File: bin/atlassian-backup (hunk 5 above)
Comment (MelonSmasher) [reply bin/atlassian-backup:77]: Did you want me to submit this as well?

**Comment by jmcgeheeiv** [comment]
Well done, @MelonSmasher.  Thank you for the enhancement.

**Comment by MelonSmasher** [comment]
> Well done, @MelonSmasher. Thank you for the enhancement.

Not a problem happy to contribute!

**Pull request merged**